src/update_uuids/create_submission_from_spreadsheet_with_uuids.py

```diff
@@ -0,0 +1,151 @@
+                process = ingest_api.create_process(submission_url, base_process)
+                global_uuid_mapping['processes'][process_uuid] = process['uuid']['uuid']
+
+
+def create_rest_entities(spreadsheet_json, ingest_api, submission):
+    submission_url = ingest_api.get_link_from_resource(submission, 'self')
+    spreadsheet_json = {'protocol': spreadsheet_json['protocol'],
+                        'biomaterial': spreadsheet_json['biomaterial'],
+                        'file': spreadsheet_json['file']}
+    for entity_type, entity_dict in spreadsheet_json.items():
+        if entity_type == 'project':
+            continue
+        for entity_uuid, value in entity_dict.items():
+            print(f'Creating {entity_type} {entity_uuid}')
+            api_function_to_call = eval(f"ingest_api.create_{entity_type}")
+            kwargs = {
+                'submission_url': submission_url,
+                'content': value.get('content', {})
+            }
+            if entity_type == 'file':
+                kwargs.update({'filename': value.get('content', {}).get('file_core', {}).get('file_name', '')})
+            entity = api_function_to_call(**kwargs)
+            global_uuid_mapping[entity_type][entity_uuid] = entity['uuid']['uuid']
+
+            for i, process_uuid in enumerate(value.get('external_links_by_entity', {}).get('process', [])):
+                process = ingest_api.get_entity_by_uuid('processes', global_uuid_mapping['processes'][process_uuid])
+                to_link = ingest_api.get_link_from_resource(process, 'protocols')
+
+                # Linking protocols
+                for prot_uuid in value.get('external_links_by_entity', {}).get('protocol', []):
+                    protocol = ingest_api.get_entity_by_uuid('protocols', global_uuid_mapping['protocol'][prot_uuid])
+                    from_link = ingest_api.get_link_from_resource(protocol, 'self')
+                    link_entities(to_link, from_link, ingest_api)
+
+                # Linking derived biomaterials
+                to_link = ingest_api.get_link_from_resource(entity, 'derivedByProcesses')
+                from_link = ingest_api.get_link_from_resource(process, 'self')
+                link_entities(to_link, from_link, ingest_api)
+
+                # Linking input biomaterials
+                input_biomaterial = ingest_api.get_entity_by_uuid('biomaterials', global_uuid_mapping['biomaterial'][value['external_links_by_entity']['biomaterial'][i]])
+                to_link = ingest_api.get_link_from_resource(input_biomaterial, 'inputToProcesses')
+                link_entities(to_link,from_link, ingest_api)
+
+
+def save_global_mapping():
+    with open('uuid_mapping.json', 'w') as f:
+        json.dump(global_uuid_mapping, f, indent=4, separators=(', ', ': '))
+
+def main(project_uuid, token, spreadsheet_path, submission_uuid):
+    ingest_api = IngestApi('https://api.ingest.archive.data.humancellatlas.org/')
```

Comment:
Don't use a hard coded ingest api url.
If you leave it blank, it is taken from the `INGEST_API` environment variable.
Add a note in the readme to set the correct variable prior to running the script.
```suggestion
    ingest_api = IngestApi()
```